packages/loader/container-loader/src/test/connectionManager.spec.ts

```diff
@@ -211,15 +228,89 @@ describe("connectionManager", () => {
 		assert.strictEqual(disconnectCount, 1, "Expect 1 disconnect from emitting a Nack");
 
 		// Async test we aren't connected within 300 ms
-		let checkedTimeout = false;
-		setTimeout(() => {
-			assert.strictEqual(connectionCount, 1, "Expect there to still not be a connection yet");
-			checkedTimeout = true;
-		}, 300);
-
+		clock.tick(300);
+		assert.strictEqual(connectionCount, 1, "Expect there to still not be a connection yet");
+		clock.tick(200);
 		connection = await waitForConnection();
 		assert.strictEqual(connectionCount, 2, "Expect there to be a connection after waiting");
-		assert(checkedTimeout, "Expected to have checked 300ms timeout");
+	});
+
+	it("Does not re-try connection on error if ReconnectMode=Disabled", async () => {
+		// mock connectToDeltaStream method so that it throws a retriable error when connect() is called in connectionManager
+		const stubbedConnectToDeltaStream = stub(mockDocumentService, "connectToDeltaStream");
+		const retryAfter = 3; // seconds
+		stubbedConnectToDeltaStream.throws(
+			// Throw retryable error
+			new RetryableError("Test message", NackErrorType.ThrottlingError, {
+				retryAfterSeconds: retryAfter,
+				driverVersion: "1",
+			}),
+		);
+		let isTimeoutSet = false;
+		const connectionManager = createConnectionManager({
+			...props,
+			// reconnectionDelayHandler should be invoked by connectionManager when the throttling errors occur causing it to attept retries
+			reconnectionDelayHandler: () => {
+				// Ideally this function from deltaManager emits "throttled" warning event which is bubbled up as container warning that host can listen to
+				// and call container.disconnect() if they wish.
+				// Emulate calling container.disconnect() which results in setting connectionManager reconnect state as "Disabled" after random amount of time
+				if (!isTimeoutSet) {
+					isTimeoutSet = true;
+					setTimeout(
+						() => {
+							connectionManager.setAutoReconnect(ReconnectMode.Disabled, {
+								text: "Container disconnected",
+							});
+						},
+						retryAfter * 1000 * 5,
+					);
+				}
+			},
+		});
+		connectionManager.connect({ text: "Test reconnect" });
+
+		await clock.tickAsync(retryAfter * 1000 * 5);
```

Comment:
maybe make delay slightly larger, to ensure we for sure will run through timeouts?